celery/contrib/django/task.py

```diff
@@ -12,10 +12,10 @@ class DjangoTask(Task):
     Provide a nicer API to trigger tasks at the end of the DB transaction.
     """
 
-    def delay_on_commit(self, *args, **kwargs):
+    def delay_on_commit(self, *args, **kwargs) -> None:
         """Call :meth:`~celery.app.task.Task.delay` with Django's ``on_commit()``."""
-        return transaction.on_commit(functools.partial(self.delay, *args, **kwargs))
```

Comment:
As we can see [in Django codebase](https://github.com/django/django/blob/ec8552417df51df8482df61b8ad78a7002634011/django/db/backends/base/base.py#L727-L750), `transaction.on_commit()` does not return anything. This `return` statement might mislead the reader in thinking that it will return something. Clearer to not have it in my opinion.